src/Apps/Artwork/ArtworkApp.tsx

```diff
@@ -150,83 +113,117 @@ export class ArtworkApp extends React.Component<Props> {
       }
       tracking.trackEvent(trackingData)
     }
-  }
-
-  render() {
-    const { artwork, me } = this.props
+  }, [artwork, tracking])
 
-    const BelowTheFoldArtworkDetails = (
-      <>
-        <Spacer mt={6} />
-        <Join separator={<Spacer mt={2} />}>
-          <ArtworkDetailsQueryRenderer slug={artwork.slug} />
+  const track = useCallback(() => {
+    trackPageview()
+    trackProductView()
+    trackLotView()
+  }, [trackPageview, trackProductView, trackLotView])
 
-          <PricingContextQueryRenderer slug={artwork.slug} />
+  const shouldRenderSubmittedOrderModal = useCallback(() => {
+    return !!props.match.location.query["order-submitted"]
+  }, [props.match.location.query])
 
-          {artwork.artists &&
-            artwork.artists.map(artist => {
-              if (!artist) return null
+  /**
+   * On mount, trigger a page view and product view
+   *
+   * FIXME: We're manually invoking pageView tracking here, instead of within
+   * the `trackingMiddleware` file as we need to pass along additional metadata.
+   * Waiting on analytics team to decide if there's a better way to capture this
+   * data that remains consistent with the rest of the app.
+   */
+  useEffect(() => {
+    if (shouldRenderSubmittedOrderModal()) {
+      // TODO: Look into using router push
+      // this.props.router.replace(this.props.match.location.pathname)
+      window.history.pushState({}, null, props.match.location.pathname)
+    }
+    track()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
 
-              return (
-                <ArtistInfoQueryRenderer key={artist.id} slug={artist.slug} />
-              )
-            })}
-        </Join>
-      </>
-    )
+  useEffect(() => {
+    if (shouldTrackPageView) {
+      track()
+    }
+  }, [shouldTrackPageView, track])
+
+  const BelowTheFoldArtworkDetails = (
+    <>
+      <Spacer mt={6} />
+      <Join separator={<Spacer mt={2} />}>
+        <ArtworkDetailsQueryRenderer slug={artwork.slug} />
+
+        <PricingContextQueryRenderer slug={artwork.slug} />
+
+        {artwork.artists &&
+          artwork.artists.map(artist => {
+            if (!artist) return null
+
+            return (
+              <ArtistInfoQueryRenderer key={artist.id} slug={artist.slug} />
+            )
+          })}
+      </Join>
+    </>
+  )
```

Comment:
I would extract this into its own component. Generally, you want to avoid declaring a component within the body of another.